conda/core/solve.py

```diff
@@ -625,7 +626,13 @@ def _remove_specs(self, ssc):
                 rec_has_a_feature = set(rec.features or ()) & feature_names
                 if rec_has_a_feature and rec.name in ssc.specs_from_history_map:
                     spec = ssc.specs_map.get(rec.name, MatchSpec(rec.name))
-                    spec._match_components.pop("features", None)
+                    spec._match_components = frozendict(
+                        {
+                            key: value
+                            for key, value in spec._match_components.items()
+                            if key != "features"
+                        }
+                    )
```

Comment:
Why do we need a frozen dict here?

Reply:
`spec._match_components` is already a `frozendict`, so the `pop` failed (the new tests exposed this)